sphinx/ext/apidoc.py

```diff
@@ -304,38 +320,43 @@ def recurse_tree(
                 # if this is not a namespace or
                 # a namespace and there is something there to document
                 if not is_namespace or has_child_module(root, excludes, opts):
-                    create_package_file(
-                        root,
-                        root_package,
-                        subpackage,
-                        files,
-                        opts,
-                        subs,
-                        is_namespace,
-                        excludes,
-                        user_template_dir,
+                    written_files.extend(
+                        create_package_file(
+                            root,
+                            root_package,
+                            subpackage,
+                            files,
+                            opts,
+                            subs,
+                            is_namespace,
+                            excludes,
+                            user_template_dir,
+                        )
                     )
                     toplevels.append(module_join(root_package, subpackage))
         else:
             # if we are at the root level, we don't require it to be a package
             assert root == rootpath
             assert root_package is None
             for py_file in files:
-                if not is_skipped_module(path.join(rootpath, py_file), opts, excludes):
+                if not is_skipped_module(Path(rootpath, py_file), opts, excludes):
                     module = py_file.split('.')[0]
-                    create_module_file(root_package, module, opts, user_template_dir)
+                    written_files.append(
+                        create_module_file(root_package, module, opts, user_template_dir)
+                    )
                     toplevels.append(module)
 
-    return toplevels
+    return written_files, toplevels
 
 
-def is_excluded(root: str, excludes: Sequence[re.Pattern[str]]) -> bool:
+def is_excluded(root: str | Path, excludes: Sequence[re.Pattern[str]]) -> bool:
     """Check if the directory is in the exclude list.
 
     Note: by having trailing slashes, we avoid common prefix issues, like
           e.g. an exclude "foo" also accidentally excluding "foobar".
     """
-    return any(exclude.match(root) for exclude in excludes)
+    root_str = Path(root).as_posix()
```

Comment:
This could potentially break configurations that were using exclude patterns with a Windows-based style I think.

Reply:
Nvm, you fixed it.

Reply:
yep already reverted